client/src/partials/projects/project.js

```diff
@@ -1,117 +1,119 @@
+        l.format = l.zone + ' - ' + l.territoire + ' (' + l.province + ')';
+      });
+      vm.zones = data;
+    }).catch(handler);
+
+    vm.state.reset();
   }
 
-  $scope.print = function print() {
-    $window.print();
-  };
-
-  $scope.new = function n() {
-    $scope.newProject = {};
-    $scope.action = 'new';
-  };
-
-  $scope.submitNew = function submitNew() {
-    var clean = connect.clean($scope.newProject);
-    connect.post('project', [clean])
-    .then(function (res) {
-      clean.id = res.data.insertId;
-      $scope.projects.post(clean);
-      $scope.action = 'default';
-    })
-    .catch(handleErrors);
-  };
-
-  $scope.cancelNew = function canceNew() {
-    $scope.newProject = {};
-  };
-
-  $scope.edit = function edit(project) {
-    $scope.editProject = angular.copy(project);
-    $scope.raw = project;
-    $scope.action = 'edit';
-  };
-
-  $scope.submitEdit = function submitEdit() {
-    var clean = connect.clean($scope.editProject);
-    connect.put('project', [clean], ['id'])
-    .then(function () {
-      $scope.projects.put(clean);
-      $scope.action = 'default';
-    })
-    .catch(handleErrors);
-  };
-
-  $scope.cancelEdit = function cancelEdit() {
-    $scope.editProject = angular.copy($scope.raw);
-    $scope.action = 'edit';
-  };
-
-  $scope.delete = function d(project) {
-    connect.delete('project', 'id', project.id)
-    .then(function () {
-      $scope.projects.remove(project.id);
+  function cancel() {
+    vm.state.reset();
+    vm.view = 'default';
+  }
+
+  function create() {
+    vm.view = 'create';
+    vm.project = {};
+  }
+
+  // Load a Project from the server
+  function loadProject(data) {
+    vm.project= data;      
+  }
+
+  // switch to update mode
+  function update(id) {
+    vm.state.reset();
+    loadProject(id);
+    vm.view = 'update';
+  }
+
+  // switch to delete warning mode
+  function del(project) {
+    vm.view = 'delete_confirm';
+    Projects.del(project.id)
+    .then(function (response) {
+      refreshProjects();
+      vm.view = 'delete_success';
+    }).catch(function (error) {
+      vm.view = 'delete_error';
+      vm.HTTPError = error;
     })
-    .catch(function (error) {
-      messenger.danger('An error occurred : ' + error);
+  }
+
+  // refresh the displayed Projects
+  function refreshProjects() {
+    return Projects.read(null, { complete : 1 })
+    .then(function (projects) {
+      vm.projects = projects.data;
     });
-  };
-
-  $scope.fmtZone = function (obj) {
-    return obj.zone + ' - ' + obj.territoire + ' (' + obj.province + ')';
-  };
-
-  appstate.register('project', function (project) {
-    $scope.currentProject = project;
-    dependencies.projects.query.where =
-      ['project.enterprise_id=' + project.enterprise_id];
-    validate.process(dependencies)
-    .then(load, handleErrors);
-  });
+  }
+
+  // form submission
+  function submit(invalid) {
+    if (invalid) { return; }
+    vm.project.locked = (vm.project.locked)?1:0;
```

Comment:
Can you make the form return the value you want using `ng-true-value` and `ng-false-value`?  [Reference Documentation](https://docs.angularjs.org/api/ng/input/input%5Bcheckbox%5D).